File: api/src/main/java/by/delaidelo/tests/testworks/domain/Result.java
package by.delaidelo.tests.testworks.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import lombok.Data;


@Entity
@Table(name = "results")
@Data
public class Result extends AbstractEntity {
    @Column
    private String fileAnalyzeDateTime; // Дата-время анализа
    @Column
    private String author; // Имя файла
    @Column
    private String fileName; // Имя файла
    @Column
    private String tempFilePath; // Путь к временному файлу
    @Column
    private Double fileSizeInMB; // Размер файла в Mb
    @Column
    private Double processingTime; // Время обработки в секундах
    @Column
    private Long count; // Кол-во валидных записей
    @Column
    private Double minValue; // Минимальное значение
    @Column
    private Double maxValue; // Максимальное значение
    @Column
    private Double average; // Среднее значение
    @Column
    private Double standardDeviation; // Стандартное отклонение
    @Column
    private Long invalidRowsCount; // Кол-во пропусков (строк без данных, с ошибкой)
    @Column
    private Long uniqueValuesCount; // Кол-во уникальных значений
}

File: ui/src/interfaces/result.ts
import { BaseDto } from "./base-dto";
import { SelectListItemDto } from "./select-list-item-dto";

export interface Result extends BaseDto {
    fileAnalyzeDateTime: string; // Дата-время анализа
    fileName: string;  // Имя файла
    tempFilePath: string; // Путь к временному файлу
    fileSizeInMB: number; // Размер файла в Mb
    processingTime: number; // Время обработки в секундах
    count: number; // Кол-во валидных записей
    minValue: number; // Минимальное значение
    maxValue: number; // Максимальное значение
    average: number; // Среднее значение
    standardDeviation: number; // Стандартное отклонение
    invalidRowsCount: number; // Кол-во пропусков (строк без данных, с ошибкой)
    uniqueValuesCount: number; // Кол-во уникальных значений
}